File: DotNetGotchas/CSharp/ExceptionInDeriving/ExceptionExample/Derived.cs
using System;

namespace ExceptionExample
{
	/// <summary>
	/// A Derived class that violates LSP.
	/// </summary>
	public class Derived : Base
	{
		/// <summary>
		/// Method1 does something with input
		/// </summary>
		/// <param name="val">val to work with</param>
		/// <exception cref="InvalidInputException">
		/// thrown if parameter is 0
		/// </exception>
		/// <exception cref="InputMustBeEvenException">
		/// thrown if parameter is not even
		/// </exception>
		public override void Method1(int val)
		{
			if ((val % 2) != 0)
			{
				// Not an even number
				throw new InputMustBeEvenException();
			}

			base.Method1(val);
			//Continue with rest of the code		
		}

	}
}

File: DotNetGotchas/VBDN/ExceptionInDeriving/ExceptionExample/Derived.vb
''' <summary>
''' A Derived class that violates LSP.
''' </summary>
Public Class Derived
    Inherits Base

    ''' <summary>
    ''' Method1 does something with input
    ''' </summary>
    ''' <param name="val">val to work with</param>
    ''' <exception cref="InvalidInputException">
    ''' thrown if parameter is 0
    ''' </exception>
    ''' <exception cref="InputMustBeEvenException">
    ''' thrown if parameter is not even
    ''' </exception>
    Public Overrides Sub Method1(ByVal val As Integer)
        If Not val Mod 2 = 0 Then
            'Not an even number
            Throw New InputMustBeEvenException
        End If


        MyBase.Method1(val)
        'Continue with rest of the code		
    End Sub

End Class